src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Trainer/TrainerDetails/TrainerDetails.cshtml.cs

```diff
@@ -0,0 +1,87 @@
+#nullable disable
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using Smart.FA.Catalog.Shared.Domain.Enumerations.Trainer;
+using Smart.FA.Catalog.Shared.Domain.Enumerations.Training;
+using Smart.FA.Catalog.Showcase.Infrastructure.Data;
+using Smart.FA.Catalog.Showcase.Web.Pages.Training.TrainingList;
+
+namespace Smart.FA.Catalog.Showcase.Web.Pages.Trainer.TrainerDetails;
+
+public class TrainerDetailsModel : PageModel
+{
+    private readonly CatalogShowcaseContext _context;
+
+    public TrainerDetailsModel(CatalogShowcaseContext context)
+    {
+        _context = context;
+    }
+
+    public TrainerDetailsViewModel Trainer { get; set; }
+
+    public async Task<IActionResult> OnGetAsync(int? id)
+    {
+        if (id == null)
+        {
+            TempData["errorMessage"] = "There is no record for this request. Please try again.";
+            return RedirectToPage("/404");
+        }
+
+        var trainerDetails = await _context.TrainerDetails.Where(trainer => trainer.Id == id).ToListAsync();
+
+        if (!trainerDetails.Any())
+        {
+            TempData["errorMessage"] = "This trainer does not exist! Please try again.";
```

Comment:
No need for the exclamation mark or try again text
```suggestion
            TempData["errorMessage"] = "This trainer does not exist.";
```